src/EStimLibrary/Core/ReusableIdPool.cs

```diff
@@ -102,13 +122,19 @@ public ReusableIdPool()
 
     public int IncrementNumIds(int increment)
     {
-        this.NumIds += increment;
-        return this.NumIds;
+        if (this.NumIds + increment <= int.MaxValue)
+        {
+            this.NumIds += increment;
+            return this.NumIds;
+        }
+            return -1;
     }
 
     public int ResetNumIds(int newMax)
     {
-        this.NumIds = newMax;
+        if (newMax < 0) this.NumIds = 0;
+        else this.NumIds = newMax;
+        this.UsedIds.RemoveWhere(id => !this.IsValidId(id));
```

Comment:
bound NumIds by the highest current used ID rather than deleting used IDs, so once NumIds set() method changes implemented, can delete this line